src/DIRAC/WorkloadManagementSystem/Agent/PushJobAgent.py

```diff
@@ -145,15 +162,21 @@ def execute(self):
             if not self._allowedToSubmit(queueName):
                 continue
 
-            # Update the configuration with the names of the Site, CE and queue to target
-            # This is used in the next stages
-            self._updateConfiguration("Site", queueDictionary["Site"])
-            self._updateConfiguration("GridCE", queueDictionary["CEName"])
-            self._updateConfiguration("CEQueue", queueDictionary["QueueName"])
-            self._updateConfiguration("RemoteExecution", True)
+            # Get a working proxy
+            ce = queueDictionary["CE"]
+            cpuTime = 86400 * 3
```

Comment:
Where does this come from?

Reply:
A proxy is needed to submit the jobs via the remote CE (ARC) apparently.
It uses a block of code from the `SiteDirector`.

The more I modify it, the more it needs to use methods from the `SiteDirector` (which is actually expected since it is a kind of "`SiteDirector` for jobs"). At some point, may be it will need to inherit from the `SiteDirector`.
What do you think?